README.md

```diff
@@ -1,138 +1,615 @@
+
+### `@metadata`
+
+The `@metdata(key, value)` decorator is similar to [`@Reflect.metadata`](https://github.com/rbuckton/reflect-metadata): It allows the easy retrival of information which was stored by annotating the class.
+
+```mjs
+import { @metadata } from "./metadata.mjs";
+
+// partially apply the decorator locally for terseness
+decorator @localMeta { @metadata("key", "value") }
+
+@localMeta class C {
+  @localMeta method() { }
+}
+
+Reflect.getMetadata(C, "key");                      // "value"
+Reflect.getMetadata(C.prototype, "key", "method");  // "value"
+```
+
+`@metadata` can also be defined in terms of `@register`. When `@register` is used with a public field, method or accessor, it is called with the second argument being the property key of that class eleemnt.
+
+```mjs
+// metadata.mjs
+import "reflect-metadata";
+
+export decorator @metadata(key, value) {
+  @register((target, prop) => Reflect.defineMetadata(key, value, target, prop))
+}
+```
+
+### `@frozen`
+
+The `@frozen` decorator freezes the constructor and its `prototype` so that they cannot be mutated after the class is defined. It leaves instances mutable. Example usage:
+
+```mjs
+import { @frozen } from "./frozen.mjs";
+
+@frozen
+class MyClass {
+  method() { }
+}
+
+MyClass.method = () => {};            // TypeError to add a method
+MyClass.prototype.method = () => {};  // TypeError to overwrite a method
+MyClass.prototype.method.foo = 1;     // TypeError to mutate a method
+```
+
+`@frozen` is implemented with the `@register` decorator, which allows a callback to be scheduled after the class is created. The callback is passed the class as an argument.
+
+```mjs
+// frozen.mjs
+export decorator @frozen {
+  @register(klass => {
+    Object.freeze(klass);
+    for (const [key, value] of Object.entries(klass)) {
+      Object.freeze(value);
+    }
+    for (const [key, value] of Object.entries(klass.prototype)) {
+      Object.freeze(value);
+    }
+  })
+}
+```
+
+### `@set`
+
+The `@set` decorator makes a class field declaration behave as a setting a property when it's called, rather than as `Object.defineProperty`. In particular, setters will be called with a normal property set, whereas `Object.defineProperty` just clobbers setters without calling them. For example:
+
+```mjs
+class SuperClass {
+  set x(value) { console.log(value); }
+}
+
+class SubClassA extends SuperClass {
+  x = 1;
+}
+
+class SubClassB extends SuperClass {
+  @set x = 1;
+}
+
+const a = new SubClassA();  // does not log anything
+a.x;                        // 1
+
+const b = new SubClassB();  // logs 1
+b.x;                        // undefined
+```
+
+The `@set` decorator is implemented with `@initialize`, which can decorate public fields. `@initialize` takes a callback as an argument, which is called after the field initializer is evaluated,
+
+```mjs
+// set.mjs
+
+export decorator @set { @initialize(function(value, key) { this[key] = value }) }
```

Comment:
does this "initialize" *replace* the define operation that would normally be performed? or has it already happened beforehand, or will it still happen after?

Reply:
It replaces the define operation. Any ideas for making that clearer in the text?

Reply:
cool, thanks for clarifying. i'm more thinking of, if we allowed multiple initializes, how would i know what the previous decorator(s) had chosen?

Reply:
I don't know; this sort of question makes me skeptical of allowing multiple initializes--I don't see a reasonable semantic for them to compose. If you have an idea, maybe propose it in an issue.